lib/elements/CheckedSelect.js

```diff
@@ -71,27 +71,35 @@ class CheckedSelect extends React.Component {
             );
         });
 
+        this._visibleOptionValues = this._visibleOptions.reduce((map, option)=>{
```

Comment:
I would not do this here since in your proposal `_visibleOptionValues` is only used in `clearVisibleOptions` function

Reply:
My suggestion is maybe to have a private helper function which returns

```
return (map, valueObject) => {
            map[valueObject[this.props.valueKey]] = true;
            return map;
        }, {})
```
that will be used to get `const valueStrings`  in **addVisibleOptions** and `const visibleOptionValues ` in **clearVisibleOptions**

Reply:
but then this will have to be recomputed each time clear is executed. It's better to only recompute it when it actually changes, when the filtering changes.